src/graphics/text.rs

```diff
@@ -107,51 +96,36 @@ impl<'s> Text<'s> {
         } else {
             Some(Text {
                 text: self.text,
-                string_length: self.string_length as u32,
+                string_length: self.string_length,
                 font: self.font
             })
         }
     }
 
-    /// Set the string of a text (from an ANSI string)
+    /// Set the string of a text
     ///
     /// A text's string is empty by default.
     ///
     /// # Arguments
     /// * string - New string
     pub fn set_string(&mut self, string: &str) {
+        let mut utf32: Vec<u32> = string.chars().map(|ch| ch as u32).collect();
```

Comment:
It's unfortunate that we have to do an allocation every time we set the string, but it's better than the currently broken behavior.

Maybe we could also offer separate APIs that make different trade-offs, for example one that takes an ascii string for those who don't need unicode strings, but want to avoid unnecessary allocations. Solving this is not the task of this PR though. This PR *does* fix previously broken functionality, so I'm merging it as-is.